service/tablet_allocator.cc

```diff
@@ -212,6 +212,8 @@ class load_balancer {
     struct shard_load {
         size_t tablet_count = 0;
 
+        std::unordered_map<table_id, size_t> tablet_count_per_table;
+
```

Comment:
Please follow up later converting the load balancer data structures to absl::flat_* where the structures are expected to be reasonable small, or the btree variants where they can be arbitrarily large. I'm worried we have large space and CPU amplification from the allocator, and since this is concentrated on shard 0, its impact is amplified.

Reply:
Fixed in V2 for new data structures.

Reply:
Something else we could have done is generated integer indexes for everything. table nr, node nr, etc. Then everything fits in plain vectors. The indexes are volatile over time, but stable for long periods and we can regenerate the data structures when they are invalidated, just as if the leader was changed. Something for later.

Reply:
btw we discussed using vectors for token metadata with @bhalevy